components/net/fetch/methods.rs

```diff
@@ -850,34 +850,25 @@ fn http_network_fetch(request: Rc<Request>,
             thread::spawn(move || {
 
                 *res_body.lock().unwrap() = ResponseBody::Receiving(vec![]);
-                let mut new_body = vec![];
-                res.response.read_to_end(&mut new_body).unwrap();
-
-                let mut body = res_body.lock().unwrap();
-                assert!(*body != ResponseBody::Empty);
-                *body = ResponseBody::Done(new_body);
-
-                // TODO: the vec storage format is much too slow for these operations,
-                // response.body needs to use something else before this code can be used
-                // *res_body.lock().unwrap() = ResponseBody::Receiving(vec![]);
-
-                // loop {
-                //     match read_block(&mut res.response) {
-                //         Ok(ReadResult::Payload(ref mut new_body)) => {
-                //             if let ResponseBody::Receiving(ref mut body) = *res_body.lock().unwrap() {
-                //                 (body).append(new_body);
-                //             }
-                //         },
-                //         Ok(ReadResult::EOF) | Err(_) => break
-                //     }
-
-                // }
-
-                // let mut completed_body = res_body.lock().unwrap();
-                // if let ResponseBody::Receiving(ref body) = *completed_body {
-                //     // TODO cloning seems sub-optimal, but I couldn't figure anything else out
-                //     *res_body.lock().unwrap() = ResponseBody::Done((*body).clone());
```

Comment:
This bit here was causing the deadlock, I've since changed the code to extract the inner body first, avoiding the deadlock.